src/pages/blog/Introduction to Callbacks, Promises and Async Await Functions in JavaScript.md

```diff
@@ -0,0 +1,175 @@
+category: "JavaScript"
+author: "Divesh Mahajan"
+layout: "@layouts/BlogLayout.astro"
+tags: [webdev, scripting, javascript, async/await, promises]
+---
+
+**About**
+
+In this blog, we will be discussing callbacks, promises and async/await functions in javascript also we will be learning about synchronous and asynchronous programming languages. If you're new to JavaScript programming, or even if you've been working with the language for a while, you may have heard these terms thrown around. Today we will be discussing each of these terms in detail.😀
+
+**What are Synchronous and Asynchronous languages?**
+
+In synchronous languages, code is executed sequentially means each line of code is executed one after another, and the program waits for each task to complete before moving on to the next one whereas asynchronous languages allow tasks to run independently and concurrently. Instead of waiting for each task to complete, the program continues executing other code while the tasks are being processed in the background.
+
+In JavaScript, code execution is typically synchronous. However, there are cases where we need to perform tasks that take time, such as making an API call or reading a file from a disk. These tasks are asynchronous because they don't block the execution of other code. So to handle asynchronous code in Javascript we use callbacks, promises and async/await functions.
+
+🚀Let's understand these concepts one by one.
+
+**Callbacks**
+
+**_Callbacks are functions that are passed as arguments to other functions and are executed at a later time or after a certain event occurs._**
+
+In JavaScript, functions can be passed as arguments to other functions. A callback is simply a function that is passed as an argument to another function and gets executed after the main function has completed its task. Let's understand callback functions from the examples.
+
+**Simple Code**
+
+function greet(name, callback) {
+console.log("Hello, " + name + "!");
+callback();
+}
+function sayGoodbye() {
+console.log("Goodbye!");
+}
+greet("Divesh", sayGoodbye);
+
+In this example, we have a function called greet that takes two arguments: name and callback. It logs a greeting message with the provided name and then executes the callback function. We also have a separate function called sayGoodbye that simply logs "Goodbye!".
+
+But one of the major drawbacks of a callback is callback hell. A Callback hell occurs when multiple asynchronous operations are nested within each other using callbacks, resulting in deeply nested and hard-to-read code. This nesting can make the code difficult to maintain, debug, and understand. It becomes challenging to handle errors and manage the flow of execution.
+
+**Example of Callback hell :**
+
+const cart = ['item1', 'item2', 'item3']; // array of items
+createOrder(cart, function (){
+proceedPayment( function (){
+showOrderSummary( function(){  
+ updateWallet()
+})
+})
+})
```

Comment:
- Add back ticks to show the codes in proper format as per Markdown syntax.

```JS
console.log("Like This");
```

- Use proper tab / spaces.